move inline js for author autocomplete

RayBB opened the pull request:
<!-- What issue does this PR close? -->
Move autocomplete part of inline js for `openlibrary/templates/books/author-autocomplete.html` #4474

<!-- What does this PR achieve? [feature|hotfix|fix|refactor] -->
refactor

### Technical
<!-- What should be noted about the implementation? -->
Nothing interesting in this PR.
Following the standards set for previous tickets in epic.

### Testing
<!-- Steps for reviewer to reproduce/verify what this PR does/fixes. -->
* Got to a [book edit page](https://openlibrary.org/books/OL7455361M/Biological_Science/edit) and try adding/removing authors 

### Screenshot
<!-- If this PR touches UI, please post evidence (screenshots) of it behaving correctly. -->

https://user-images.githubusercontent.com/921217/118733787-adb3e900-b7d8-11eb-9873-ce111a60f205.mp4


### Stakeholders
<!-- @ tag stakeholders of this bug -->
@jdlrobson

jdlrobson commented on openlibrary/templates/books/author-autocomplete.html:
```diff
@@ -38,28 +38,8 @@
         <a href="javascript:;" class="add">$_("Add another author?")</a>
     </div>
-
-<div id="authors">
```
MINOR UI REGRESSION:

Removing this ID  causes the cross icon to jump to a new line for me. Let's make sure the rule `#authors div, div.lists` still applies - potentially by adding `multi-input-autocomplete--author` to static/css/page-user.less for now.


<img width="745" alt="Screen Shot 2021-05-23 at 10 25 35 AM" src="https://user-images.githubusercontent.com/148752/119270371-38b62a00-bbb1-11eb-9645-5060b64ac776.png">

jdlrobson commented on openlibrary/plugins/openlibrary/js/index.js:
```diff
@@ -93,25 +104,46 @@ jQuery(function () {
+    ) {
         import(/* webpackChunkName: "user-website" */ './edit')
-            .then(module => module.initWorksMultiInputAutocomplete());
+            .then(module => {
```
FYI: I merged into a single entry point here.

jdlrobson commented on openlibrary/plugins/openlibrary/js/edit.js:
```diff
@@ -72,6 +75,29 @@ export function initWorksMultiInputAutocomplete() {
+            {
+                endpoint: '/authors/_autocomplete',
+                // Don't render "Create new author" if searching by key
+                addnew: function(query) { return !/^OL\d+A/i.test(query); },
```
Consider using shorthand e.g. `(query)=> !/^OL\d+A/i.test(query)`

jdlrobson commented on openlibrary/plugins/openlibrary/js/index.js:
```diff
@@ -1,3 +1,14 @@
+        $('input#work').val(`/works/${li.attr('id')}`);
+        $('form#addbook').submit();
+    });
+    $('#bookAddCont').click(function() {
```
FYI (for future patches): https://github.com/internetarchive/openlibrary/issues/5198

RayBB replied:
I think you accidentally added that function. I removed it

jdlrobson replied:
Thanks!

jdlrobson reviewed (dismissed):
UI regression is the only remaining blocker to merging this one.

RayBB commented:
@jdlrobson good catch on the visual regression. Went ahead and changed the .less file to use the class and it works great now. Also addressed you other comments.
